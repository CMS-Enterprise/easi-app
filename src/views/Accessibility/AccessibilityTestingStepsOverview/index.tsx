import React from 'react';
import { Trans, useTranslation } from 'react-i18next';
import { Link, useLocation } from 'react-router-dom';
import { Link as UswdsLink } from '@trussworks/react-uswds';

import BreadcrumbNav from 'components/BreadcrumbNav';
import PageHeading from 'components/PageHeading';
import CollapsableLink from 'components/shared/CollapsableLink';
import { Step, StepBody, StepHeading, StepList } from 'components/StepList';

import './index.scss';

const AccessibilityTestingStepsOverview = () => {
  const { t } = useTranslation('accessibility');
  const location = useLocation();
  const params = new URLSearchParams(location.search);
  const exceptionReasons: string[] = t(
    'testingStepsOverview.exception.reasons',
    {
      returnObjects: true
    }
  );

  return (
    <div className="grid-container">
      <div className="tablet:grid-col-10">
        <BreadcrumbNav>
          <li>
            <Link to="/">Home</Link>
            <i className="fa fa-angle-right margin-x-05" aria-hidden />
          </li>
          <li>Steps Involved in 508 testing</li>
        </BreadcrumbNav>
        <PageHeading>{t('testingStepsOverview.heading')}</PageHeading>
        <p className="accessibility-testing-overview__description">
          {t('testingStepsOverview.description')}
        </p>
      </div>
      <div className="tablet:grid-col-6 margin-top-6">
        <StepList>
          <Step>
            <StepHeading>
              {t('testingStepsOverview.fillForm.heading')}
            </StepHeading>
            <StepBody>
              <p className="margin-0">
                {t('testingStepsOverview.fillForm.description')}
              </p>
            </StepBody>
          </Step>
          <Step>
            <StepHeading>
              {t('testingStepsOverview.prepareVPAT.heading')}
            </StepHeading>
            <StepBody>
              <Trans
                i18nKey="accessibility:testingStepsOverview.prepareVPAT.fillOutVPAT"
                className="margin-0"
              >
                indexZero
                <UswdsLink
                  href="/508/templates"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  templatesLink
                </UswdsLink>
                indexTwo
              </Trans>
              <p className="margin-bottom-0">
                {t('testingStepsOverview.prepareVPAT.changesVPAT')}
              </p>
            </StepBody>
          </Step>
          <Step>
            <StepHeading>
              {t('testingStepsOverview.testingSession.heading')}
            </StepHeading>
            <StepBody>
              <p className="margin-0">
                {t('testingStepsOverview.testingSession.description')}
              </p>
            </StepBody>
          </Step>
          <Step>
            <StepHeading>
              {t('testingStepsOverview.results.heading')}
            </StepHeading>
            <StepBody>
              <p className="margin-0">
                {t('testingStepsOverview.results.description')}
              </p>
              <dl title="508 test scores">
                <dt className="text-bold margin-bottom-1">
                  {t('testingStepsOverview.results.score.above99.heading')}
                </dt>
                <dd className="margin-left-0 margin-bottom-2">
                  {t('testingStepsOverview.results.score.above99.description')}
                </dd>
                <dt className="text-bold margin-bottom-1">
                  {t('testingStepsOverview.results.score.interval75.heading')}
                </dt>
                <dd className="margin-left-0 margin-bottom-2">
                  {t(
                    'testingStepsOverview.results.score.interval75.description'
                  )}
                </dd>
                <dt className="text-bold margin-bottom-1">
                  {t('testingStepsOverview.results.score.below75.heading')}
                </dt>
                <dd className="margin-left-0">
                  {t('testingStepsOverview.results.score.below75.description')}
                </dd>
              </dl>
            </StepBody>
          </Step>
        </StepList>
        <CollapsableLink
          id="easi-508-testing-exception"
          label={t('testingStepsOverview.exception.label')}
          styleLeftBar={false}
        >
          <div className="line-height-body-5">
            <p className="margin-top-0">
              {t('testingStepsOverview.exception.description')}
            </p>
            <ul>
              {exceptionReasons.map(reason => (
                <li key={reason}>{reason}</li>
              ))}
            </ul>
            <p>{t('testingStepsOverview.exception.exceptionFineprint')}</p>
            <p>
              <Trans i18nKey="accessibility:testingStepsOverview.exception.contact">
                indexZero
                <UswdsLink href="mailto:CMS_Section508@cms.hhs.gov">
                  email
                </UswdsLink>
                indexTwo
              </Trans>
            </p>
          </div>
        </CollapsableLink>
        {params.get('continue') === 'true' && (
          <UswdsLink
            asCustom={Link}
            className="usa-button margin-top-8"
            variant="unstyled"
            to="/508/requests/new"
            data-testid="continue-link"
          >
            Get started with Step 1
          </UswdsLink>
        )}
      </div>
    </div>
  );
};

export default AccessibilityTestingStepsOverview;
