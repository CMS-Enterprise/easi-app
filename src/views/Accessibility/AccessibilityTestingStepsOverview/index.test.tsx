import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import AccessibilityTestingStepsOverview from './index';

describe('The accessibility testing overview', () => {
  it('renders without crashing', () => {
    shallow(
      <MemoryRouter>
        <AccessibilityTestingStepsOverview />
      </MemoryRouter>
    );
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <AccessibilityTestingStepsOverview />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
