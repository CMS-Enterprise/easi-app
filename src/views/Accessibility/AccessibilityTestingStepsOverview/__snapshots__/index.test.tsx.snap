// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The accessibility testing overview matches the snapshot 1`] = `
<div
  className="grid-container"
>
  <div
    className="tablet:grid-col-10"
  >
    <nav
      aria-label="breadcrumbs"
      className="easi-breadcrumb-nav"
      role="navigation"
    >
      <ol>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <i
            aria-hidden={true}
            className="fa fa-angle-right margin-x-05"
          />
        </li>
        <li>
          Steps Involved in 508 testing
        </li>
      </ol>
    </nav>
    <h1
      aria-live="polite"
      className="easi-h1"
      tabIndex={-1}
    >
      Steps involved in 508 testing
    </h1>
    <p
      className="accessibility-testing-overview__description"
    >
      Here is an overview of the 508 process for testing your application.
    </p>
  </div>
  <div
    className="tablet:grid-col-6 margin-top-6"
  >
    <ol
      className="easi-step-list"
    >
      <li
        className="easi-step"
      >
        <div
          className="margin-left-1 margin-bottom-3 tablet:margin-bottom-4"
        >
          <h3
            className="margin-top-0 margin-bottom-1"
          >
            Fill the request form in EASi
          </h3>
          <div
            className="line-height-body-4"
          >
            <p
              className="margin-0"
            >
              Tell the 508 team which application you plan to test.
            </p>
          </div>
        </div>
      </li>
      <li
        className="easi-step"
      >
        <div
          className="margin-left-1 margin-bottom-3 tablet:margin-bottom-4"
        >
          <h3
            className="margin-top-0 margin-bottom-1"
          >
            Prepare and upload the VPAT and Test plan
          </h3>
          <div
            className="line-height-body-4"
          >
            Download and fill the VPAT and Test plan from the 
            <a
              className="usa-link"
              href="/508/templates"
              rel="noopener noreferrer"
              target="_blank"
            >
              templates page (opens in a new tab)
            </a>
            . These documents will help the 508 team prepare for testing. Uploaded your completed documents to EASi for the 508 team to review.
            <p
              className="margin-bottom-0"
            >
              The 508 team will get back to you via email about any changes needed prior to testing.
            </p>
          </div>
        </div>
      </li>
      <li
        className="easi-step"
      >
        <div
          className="margin-left-1 margin-bottom-3 tablet:margin-bottom-4"
        >
          <h3
            className="margin-top-0 margin-bottom-1"
          >
            Attend the testing session
          </h3>
          <div
            className="line-height-body-4"
          >
            <p
              className="margin-0"
            >
              The 508 team will work with you to schedule a testing session and you will test your application together. Depending on the results, you may need to address any issues and retest.
            </p>
          </div>
        </div>
      </li>
      <li
        className="easi-step"
      >
        <div
          className="margin-left-1 margin-bottom-3 tablet:margin-bottom-4"
        >
          <h3
            className="margin-top-0 margin-bottom-1"
          >
            Receive results
          </h3>
          <div
            className="line-height-body-4"
          >
            <p
              className="margin-0"
            >
              Your test score will determine your next steps:
            </p>
            <dl
              title="508 test scores"
            >
              <dt
                className="text-bold margin-bottom-1"
              >
                99% and above
              </dt>
              <dd
                className="margin-left-0 margin-bottom-2"
              >
                508 testing is complete and you can release your application.
              </dd>
              <dt
                className="text-bold margin-bottom-1"
              >
                Between 99% and 75%
              </dt>
              <dd
                className="margin-left-0 margin-bottom-2"
              >
                You can release your application but you will need fix the issues and retest within a year. You need to submit a remediation plan as a part of retesting.
              </dd>
              <dt
                className="text-bold margin-bottom-1"
              >
                75% and below
              </dt>
              <dd
                className="margin-left-0"
              >
                You cannot release your application. You need to fix all issues right away and retest.
              </dd>
            </dl>
          </div>
        </div>
      </li>
    </ol>
    <div
      className="easi-collapsable-link"
    >
      <button
        aria-controls="easi-508-testing-exception"
        aria-expanded={false}
        className="usa-button usa-button--unstyled"
        data-testid="button"
        onClick={[Function]}
        type="button"
      >
        <span
          className="fa easi-collapsable-link__square fa-caret-right"
        />
        What if I need an exception from 508 testing?
      </button>
    </div>
  </div>
</div>
`;
